### DISCUSSION TITLE
Critical Dependency Warning

### DISCUSSION THREAD

**Discussion by Maanil-Verma-WIN** [harttle/liquidjs#718] [Q&A]
I am working with NextJS 13, Material UI, and TypeScript & I am getting this warning in my VS Code Terminal Again and again.

```
warn  - ./node_modules/liquidjs/dist/liquid.node.cjs.js
**Critical dependency: require function is used in a way in which dependencies cannot be statically extracted**

Import trace for requested module:
./node_modules/liquidjs/dist/liquid.node.cjs.js
./shared-components/ComponentA.tsx
./shared-components/ComponentB.tsx
./shared-components/ComponentC.tsx
./shared-components/ComponentD.tsx
./app/components/(dashboard-app)/dashboard/components/ComponentE.tsx
./app/component/(dashboard-app)/dashboard/ComponentF.tsx
```

**Comment by harttle** [comment]
It's related to a functionality to load partial templates via `require.resolve`, we'll have an option to remove this functionality in the next major release. Need more discussion on this.

There's a way to mitigate the warning from [stackoverflow](https://stackoverflow.com/questions/59235497/i-get-a-webpack-warning-critical-dependency-the-request-of-a-dependency-is-an-e), add a plugin to Webpack config:

```javascript
  plugins: [
    new ContextReplacementPlugin(/liquidjs/)
  ],
```

See [a demo config here](https://github.com/harttle/liquidjs/blob/e401a30972c70be501d91fd999dc305ec67a15a2/demo/webpack/webpack.config.js#L8).
